bundles/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/internal/handler/FroniusBaseThingHandler.java

```diff
@@ -149,4 +165,52 @@ protected void updateChannel(String channelId) {
      * @param bridgeConfiguration the connected bridge configuration
      */
     public abstract void refresh(FroniusBridgeConfiguration bridgeConfiguration);
+
+    /**
+     *
+     * @param type response class type
+     * @param url to request
+     * @return the object representation of the json response
+     */
+    protected <T extends BaseFroniusResponse> T collectDataFormUrl(Class<T> type, String url) {
+        T result = null;
+        boolean resultOk = false;
+        String errorMsg = null;
+
+        try {
+            logger.debug("URL = {}", url);
+            String response = HttpUtil.executeUrl("GET", url, API_TIMEOUT);
+
+            if (response != null) {
+                logger.debug("aqiResponse = {}", response);
+                result = gson.fromJson(response, type);
+            }
+
+            if (result == null) {
+                errorMsg = "no data returned";
+            } else {
+                if (result.getHead().getStatus().getCode() == 0) {
+                    resultOk = true;
+                } else {
+                    errorMsg = result.getHead().getStatus().getReason();
+                }
+            }
+            if (!resultOk) {
+                logger.debug("Error in fronius response: {}", errorMsg);
+            }
+        } catch (JsonSyntaxException e) {
+            errorMsg = "Configuration is incorrect";
+            logger.debug("Error running fronius request: {}", errorMsg);
+        } catch (IOException | IllegalStateException e) {
+            logger.debug("Error running fronius request: {}", e.getMessage());
+        }
```

Comment:
Shouldn't you fill out the `errorMsg` here?

Reply:
This was original code, just moved to this class. Nevertheless, good catch. I have assigned e.getMessage() to errorMsg.